### PR TITLE
[FEAT] Refactor Collectible Buff Labels

### PR DESCRIPTION
**Pull request by eliasSFL**
# Description

I refactored `COLLECTIBLE_BUFF_LABELS` so that we can allow for conditional descriptions based on skills that changes the buffs of another item:

For example
```typescript
  "Immortal Pear": {
    shortDescription: state.bumpkin.skills["Pear Turbocharge"]
      ? translate("description.immortal.pear.boosted.boost")
      : translate("description.immortal.pear.boost"),
    labelType: "success",
    boostTypeIcon: powerup,
  },
  ```

As a result of this change, every place this functional object is called, will have to be updated

Fixes #issue

# What needs to be tested by the reviewer?

Please describe how this can be tested.

# Checklist:

- [ ] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [ ] Screenshot if it includes any UI changes
- [ ] I have read the contributing guidelines and agree to the T&Cs
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by eliasSFL** [comment]
This PR is to be merged after #4894

File: src/features/island/collectibles/components/Hourglass.tsx [hunk 1]
```diff
@@ -177,6 +181,11 @@ export const Hourglass: React.FC<HourglassProps> = ({
   hourglass,
 }) => {
   const { gameService, showTimers } = useContext(Context);
+  const [
```
Comment (spencerdezartsmith) [inline src/features/island/collectibles/components/Hourglass.tsx:184]: It is a much better practice to implement `useSelector` here with a callback to get access just to the `state` value only. This follows the best practice recommendation of _least privilege - only accessing what you need_. Using `useActor` subscribes this component to ALL state changes in the machine which will then cause a re-render. We should always try to access only the data we need. I recommend refactoring all of these actors to selectors instead:

```ts
// Outside component
const _state = (state: MachineState) => state.context.state

// Inside component
const state = useSelector(gameService, _state)
```

File: src/features/island/hud/components/codex/components/Detail.tsx [hunk 2]
```diff
@@ -44,6 +46,12 @@ export const Detail: React.FC<Props> = ({
     availability,
   } = ITEM_DETAILS[name];
   const [imageWidth, setImageWidth] = React.useState<number>(0);
+  const { gameService } = useContext(Context);
+  const [
```
Comment (spencerdezartsmith) [inline src/features/island/hud/components/codex/components/Detail.tsx:50]: Same here

File: src/features/marketplace/components/Collection.tsx [hunk 3]
```diff
@@ -27,6 +28,12 @@ export const Collection: React.FC<{
   search?: string;
   onNavigated?: () => void;
 }> = ({ search, onNavigated }) => {
+  const { gameService } = useContext(Context);
```
Comment (spencerdezartsmith) [inline src/features/marketplace/components/Collection.tsx:31]: Same here

File: src/features/marketplace/components/PurchaseModalContent.tsx [hunk 4]
```diff
@@ -39,6 +39,11 @@ export const PurchaseModalContent: React.FC<PurchaseModalContentProps> = ({
   const { gameService } = useContext(Context);
   const { openModal } = useContext(ModalContext);
   const { t } = useAppTranslation();
+  const [
```
Comment (spencerdezartsmith) [inline src/features/marketplace/components/PurchaseModalContent.tsx:42]: Same here

File: src/features/marketplace/components/RemoveListing.tsx [hunk 5]
```diff
@@ -26,6 +26,11 @@ export const RemoveListing: React.FC<Props> = ({
 }) => {
   const { t } = useAppTranslation();
   const { gameService } = useContext(Context);
+  const [
```
Comment (spencerdezartsmith) [inline src/features/marketplace/components/RemoveListing.tsx:29]: Same here

File: src/features/marketplace/components/RemoveOffer.tsx [hunk 6]
```diff
@@ -25,6 +26,11 @@ export const RemoveOffer: React.FC<Props> = ({
 }) => {
   const { t } = useAppTranslation();
   const { gameService } = useContext(Context);
+  const [
```
Comment (spencerdezartsmith) [inline src/features/marketplace/components/RemoveOffer.tsx:29]: Same here

File: src/features/marketplace/components/TopTrades.tsx [hunk 7]
```diff
@@ -10,13 +10,19 @@ import { NPC } from "features/island/bumpkin/components/NPC";
...
 import { Context } from "features/game/GameProvider";
+import { useActor } from "@xstate/react";
 
 export const TopTrades: React.FC<{
   trends?: MarketplaceTrends;
 }> = ({ trends }) => {
   const { t } = useAppTranslation();
   const navigate = useNavigate();
   const { gameService } = useContext(Context);
+  const [
```
Comment (spencerdezartsmith) [inline src/features/marketplace/components/TopTrades.tsx:21]: Same here

File: src/features/marketplace/components/TrendingTrades.tsx [hunk 8]
```diff
@@ -13,14 +13,19 @@ import decreaseIcon from "assets/icons/decrease_arrow.png";
...
+import { useActor } from "@xstate/react";
 
 export const TrendingTrades: React.FC<{
   trends?: MarketplaceTrends;
 }> = ({ trends }) => {
   const { t } = useAppTranslation();
   const isWorldRoute = useLocation().pathname.includes("/world");
-
   const { gameService } = useContext(Context);
+  const [
```
Comment (spencerdezartsmith) [inline src/features/marketplace/components/TrendingTrades.tsx:24]: You get the idea 😉

**Review by spencerdezartsmith — CHANGES_REQUESTED** [review]
Nice one! Just some changes around accessing `state`.

**Comment by eliasSFL** [comment]
@spencerdezartsmith I've used your suggestions. 

So as much as possible I should just call the state using useSelector if I'm only accessing items in the GameState?

**Review by spencerdezartsmith — APPROVED** [review]
LGTM 👏🏼

**Pull request merged**